src/main/java/core/basesyntax/service/impl/ReportMakerImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ReportMaker;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ReportMakerImpl implements ReportMaker {
+    private static final String HEAD_OF_REPORT = "fruit,quantity";
+    private static final String COMMA = ",";
+
+    @Override
+    public Map<String, Long> amountCalculator(List<Fruit> fruits) {
+        return fruits.stream()
+                .map(Fruit::getName)
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+    }
+
+    @Override
+    public String makeReport(List<Fruit> fruits) {
+        Map<String, Long> totalAmount = amountCalculator(fruits);
```

Comment:
Not a mistake, but question to mentors.
Is it ok to use long numbers where we can use ints?